git_machete/cli.py

```diff
@@ -75,38 +78,172 @@ def usage(command: str = None) -> None:
+
+    subparsers.add_parser('hooks')
+
+    is_managed_parser = subparsers.add_parser('is-managed')
+    is_managed_parser.add_argument('branch', nargs='?')
+
+    list_parser = subparsers.add_parser('list')
+    list_parser.add_argument(
+        'category',
+        choices=[
+            'addable', 'managed', 'slidable', 'slidable-after', 'unmanaged',
+            'with-overridden-fork-point']
+    )
+    list_parser.add_argument('branch', nargs='?', default=argparse.SUPPRESS)
+
+    log_parser = subparsers.add_parser('log', aliases=['l'])
+    log_parser.add_argument('branch', nargs='?', default=argparse.SUPPRESS)
+
+    reapply_parser = subparsers.add_parser('reapply')
+    reapply_parser.add_argument('-f', '--fork-point', default=argparse.SUPPRESS)
+
+    show_parser = subparsers.add_parser('show')
+    show_parser.add_argument(
+        'direction',
+        choices=['c', 'current', 'd', 'down', 'f', 'first', 'l', 'last',
+                 'n', 'next', 'p', 'prev', 'r', 'root', 'u', 'up']
+    )
+    show_parser.add_argument('branch', nargs='?', default=argparse.SUPPRESS)
+
+    slide_out_parser = subparsers.add_parser('slide-out', argument_default=argparse.SUPPRESS)
+    slide_out_parser.add_argument('branch', nargs='+')
+    slide_out_parser.add_argument('-d', '--down-fork-point')
+    slide_out_parser.add_argument('-M', '--merge', action='store_true')
+    slide_out_parser.add_argument('-n', action='store_true')
+    slide_out_parser.add_argument('--no-edit-merge', action='store_true')
+    slide_out_parser.add_argument('--no-interactive-rebase', action='store_true')
+
+    squash_parser = subparsers.add_parser('squash')
+    squash_parser.add_argument('-f', '--fork-point', default=argparse.SUPPRESS)
+
+    status_parser = subparsers.add_parser(
+        'status', aliases=['s'], argument_default=argparse.SUPPRESS)
+    status_parser.add_argument('--color', choices=['always', 'auto', 'never'], default='auto')
+    status_parser.add_argument('-l', '--list-commits', action='store_true')
+    status_parser.add_argument('-L', '--list-commits-with-hashes', action='store_true')
+    status_parser.add_argument('--no-detect-squash-merges', action='store_true')
+
+    traverse_parser = subparsers.add_parser('traverse', argument_default=argparse.SUPPRESS)
+    traverse_parser.add_argument('-F', '--fetch', action='store_true')
+    traverse_parser.add_argument('-l', '--list-commits', action='store_true')
+    traverse_parser.add_argument('-M', '--merge', action='store_true')
+    traverse_parser.add_argument('-n', action='store_true')
+    traverse_parser.add_argument('--no-edit-merge', action='store_true')
+    traverse_parser.add_argument('--no-interactive-rebase', action='store_true')
+    traverse_parser.add_argument('--no-detect-squash-merges', action='store_true')
+    traverse_parser.add_argument('--push', action='store_true')
+    traverse_parser.add_argument('--no-push', action='store_true')
```

Comment:
Hmm can it be sth like `store_false` but for `--push` flag? since that's basically the current logic

Reply:
Few lines up, there is for whole subparser default=argparse.SUPPRESS which causes particular argument not to be even in the argparse.Namespace if it was not in the parsed arguments. This way if I will pass this optional argument it will show as push=True, otherwise it will not show at all.

If I implement it with 'store_false' and pass this argument it will show as push=False, so it's opposite of what we want. If I won't pass it it will show as push=True.

https://docs.python.org/3/library/argparse.html#action